OpenMarket/OpenMarket/ViewController.swift

```diff
@@ -6,13 +6,229 @@
 
 import UIKit
 
+enum ArrangeMode: Int {
+    case list = 0
+    case grid = 1
+}
+
 class ViewController: UIViewController {
+    private let segmentedControl = UISegmentedControl(items: ["LIST", "GRID"])
+    private var arrangeMode: ArrangeMode = .list
+    private var data: [Product] = []
+    private lazy var collectionView = UICollectionView(frame: .zero, collectionViewLayout: listLayout())
+    private lazy var activityIndicator: UIActivityIndicatorView = {
+        createActivityIndicator()
+    }()
+}
 
+// MARK: - Life Cycle
+extension ViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        setUpNavigationItems()
+        self.view.addSubview(collectionView)
+        self.view.addSubview(activityIndicator)
+        self.activityIndicator.startAnimating()
+        receiveProductListData()
+        
+        setUpSegmentedControlLayout()
+        setUpCollectionViewConstraints()
+        collectionViewDelegate()
+        
+        setUpInitialState()
     }
+}
 
+// MARK: - Delegate
+extension ViewController {
+    func collectionViewDelegate() {
+        collectionView.delegate = self
+        collectionView.dataSource = self
+    }
+}
 
+// MARK: - Delegate Method
+extension ViewController: UICollectionViewDelegate, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        data.count
```

Comment:
메서드에서는 return 키워드 생략은 잘하지 않는 것 같습니다.
```suggestion
        return data.count
```

Reply:
깜빡했네요 😂 수정했습니다.

Reply:
return 붙여주었습니다.